internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -295,18 +295,18 @@ func resourceApiManagementApiCreateUpdate(d *pluginsdk.ResourceData, meta interf
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewApiID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), d.Get("name").(string))
-
 	revision := d.Get("revision").(string)
 	path := d.Get("path").(string)
-	apiId := fmt.Sprintf("%s;rev=%s", id.Name, revision)
+	apiId := fmt.Sprintf("%s;rev=%s", d.Get("name").(string), revision)
```

Comment:
how come id.Name doesn't work here?

Reply:
Not specifying an exact revision as part of the tracked ID will lead to an unstable state, due to the way the ARM API works.
As such, I define `apiID` locally and moved the call to `parse.NewApiID` afterwards to use this `apiID`

Reply:
since this is a Required field, users will always have to specify a Revision, so can you explain the problem?